plugins/modules/api.py

```diff
@@ -304,6 +308,11 @@ def __init__(self):
             update=dict(type='str'),
             cmd=dict(type='str'),
             query=dict(type='str'),
+            extended_query=dict(type='dict', options=dict(
+                attributes=dict(type='list', required=True),
+                where=dict(type='list')
```

Comment:
This way Ansible will do most of the validation:
```suggestion
                where=dict(
                    type='list',
                    elements='dict',
                    options=dict(
                        attribute=dict(type='str'),
                        is=dict(type='str', choices=["==", "!=", ">", "<", "in", "eq", "not", "more", "less"]),
                        value=dict(type='raw'),
                        or=dict(type='list', elements='dict', options=dict(
                            attribute=dict(type='str', required=True),
                            is=dict(type='str', choices=["==", "!=", ">", "<", "in", "eq", "not", "more", "less"], required=True),
                            value=dict(type='raw', required=True),
                        )),
                    ),
                    required_together=[('attribute', 'is', 'value')],
                    mutually_exclusive=[('attribute', 'or')],
                    required_one_of=[('attribute', 'or')],
                ),
```
The only thing missing are the checks for `value`'s type (depending on whether `is` is `in` or not).